health/health_json.c

```diff
@@ -142,6 +142,37 @@ static inline void health_rrdcalc2json_nolock(RRDHOST *host, BUFFER *wb, RRDCALC
     char value_string[100 + 1];
     format_value_and_unit(value_string, 100, rc->value, rc->units, -1);
 
+    char *replaced_info = NULL;
+
+    if (likely(rc->info)) {
+        char *m;
+        replaced_info = strdupz(rc->info);
+
+        while ( m = strstr (replaced_info, "$this") ) {
```

Comment:
Hello @MrZammler ,

A huge improvement :) .
I was studying your code, and I agree 100% that it is good to have the `while`, because users can set different `$this` in the message, but I will give another suggestion for you. When the `$this` is overwritten, we do not expect to have more `$this` before the address `m`, instead to pass as argument for `strstr`  the variable `replace_info`,  I think we can pass `m`. Of course, this implies that m needs to be initialize before any call for the loop. What do you think?

Best regards!

Reply:
Yes! I was thinking about this too (honest :-) ), I will have a go I think!

I really appreciate your feedback, many thanks!

Reply:
You are welcome, I am happy to help you! :)